packages/tailwindcss/src/compat/apply-compat-hooks.ts

```diff
@@ -129,23 +126,17 @@ export async function applyCompatibilityHooks({
       return resolveThemeVariableValue(path)
     }
 
-    // Extract an eventual modifier from the path. e.g.:
-    // - "colors.red.500 / 50%" -> "50%"
-    let lastSlash = path.lastIndexOf('/')
-    let modifier: string | null = null
-    if (lastSlash !== -1) {
-      modifier = path.slice(lastSlash + 1).trim()
-      path = path.slice(0, lastSlash).trim() as ThemeKey
-    }
-
-    let themeValue = lookupThemeValue(designSystem.theme, path)
-
-    // Apply the opacity modifier if present
-    if (modifier && themeValue) {
-      return withAlpha(themeValue, modifier)
-    }
-
-    return themeValue
+    // If the theme value is not found in the simple resolver, we upgrade to the full backward
+    // compatibility support implementation of the `resolveThemeValue` function.
+    upgradeToFullPluginSupport({
```

Comment:
We should have a guard on the design system so we only do this one time. Right now for _any_ missed value this will do a good bit of additional work which ideally would only happen one time.

Also, right now there are some things that are not idempotent so you'd end up with multiple sets of the same keyframes in your theme depending on how many times you called the theme function with a non-CSS key path.

Reply:
@thecrypticace right now this only ever can run one time because `upgradeToFullPluginSupport` is sync and will replace the resolver immediately. In an earlier version I added an awkward symbol tag to the `designsystem` to be sure but if it's not possible we decided to remove it (I made `upgradeToFullPluginSupport` blocking for this very reason).

Do you still think we want an additional safety switch in case we make some changes to the logic?

Reply:
Ah yep you're totally right — this is fine then.